### ISSUE TITLE
Incorrect conversion of symbols

### ISSUE THREAD

**Issue by dlf57** [openbabel/openbabel#2333]
If a chlorine atom is represented as CL instead of Cl in a file Open Babel, does not properly convert the symbol. This can easily be seen when writing the SMILES out for the file.

- [x] I believe this to be a bug with Open Babel
- [ ] This is a feature request

## Environment Information
Open Babel version: v3.1.1 conda-forge
Operating system and version: Manjaro 20.2.1, Python 3.8

## Expected Behavior

SMILES: CCl

## Actual Behavior

SMILES: [CH3].*
## Steps to Reproduce

Writing the SMILES string for the molecule below resulted in the [CH3].* SMILES output.
```
from openbabel import pybel

mol = next(pybel.readfile("xyz", "ccl.xyz"))
print(mol.write("smi")
```
5

C         -3.25634        0.95398        0.05312
CL        -1.48936        0.95398        0.05311
H         -3.61336        1.82622        0.60505
H         -3.61337        0.03988        0.53253
H         -3.61338        0.99586       -0.97822

**Comment by ghutchis** [comment]
The code in `src/elements.cpp` GetAtomicNum() was rewritten to a set of switch statements - clearly many cases with capitalization need to be fixed (e.g., https://github.com/openbabel/openbabel/blob/master/src/elements.cpp#L257)

**Comment by Arraxx** [comment]
Can i take this issue? @ghutchis

**Comment by baoilleach** [comment]
The switch statement is generated by a Python script in the maintenance
repo (https://github.com/openbabel/maintenance/tree/master/scripts/elements).
A simple change should generate the uppercase versions.

**Comment by ghutchis** [comment]
I think we should be a bit cautious considering PDB parsing and others sometimes have weird atom naming (e.g., CA != calcium).

I suggested to @dlf57 that we target common, unambiguous cases (e.g., CL = Cl, BR = Br, etc.)

I'm open to debate about enabling all upper-case two-letter parsing. For example is CA **always** calcium?

I suppose we can go for it and target the bug reports that come later.

**Comment by baoilleach** [comment]
Or if a specific format needs to handle uppercase, e.g. xyz, it should
handle it itself. IMO, this function should focus on converting atomic
symbols (Ca, not ca or CA), and avoid the particulars of parsing various
file formats.

**Comment by ghutchis** [comment]
Yeah, that's probably the better approach. XYZ can call tolower() on the 2nd character.

**Comment by its-kunal-ssi** [comment]
is this issue still available.
